tamr_toolbox/project/_common/movement.py

```diff
@@ -0,0 +1,178 @@
+    logging.info(f"Starting to export {project.name} or type {project.type}.")
+    response = project.client.post(
+        f"/api/versioned/v1/projects/{project.resource_id}:export", json=body
+    )
+    # Raise error if export api call not successful
+    if not response.ok:
+        error_message = f"Error exporting project artifacts: {response.json()['message']}"
+        logging.error(error_message)
+        raise ValueError(error_message)
+
+    # get operation
+    job_id = response.json()["id"]
+    operation = utils.operation.from_resource_id(tamr=project.client, job_id=job_id)
+    if not asynchronous:
+        # periodically check export job until completion
+        logging.info(f"Waiting for export to be created.")
+        operation = operation.wait()
+        utils.operation.enforce_success(operation)
+        logging.info(f"Export complete: {operation}.")
+    else:
+        logging.info(f"Export called asynchronously: {operation}.")
+
+    return operation
+
+
+def import_artifacts(
+    *,
+    project_artifact_path: str,
+    tamr_client: Client,
+    target_project: Project = None,
+    new_project_name: str = None,
+    new_unified_dataset_name: Optional[str] = None,
+    exclude_artifacts: Optional[List[str]] = None,
+    include_additive_artifacts: Optional[List[str]] = None,
+    include_destructive_artifacts: Optional[List[str]] = None,
+    fail_if_not_present: bool = False,
+    asynchronous: bool = False,
+    overwrite_existing: bool = False,
+) -> Operation:
+    """Import project artifacts into a tamr instance
+
+    Version:
+        Requires Tamr 2021.005.0 or later
+
+    Args:
+        tamr_client: a tamr client
+        project_artifact_path: project artifacts zip filepath
+        target_project: an optional target project for migration
+        new_project_name: new project name
+        new_unified_dataset_name: new unified dataset name
+        exclude_artifacts: list of artifacts to exclude in import
+        include_additive_artifacts: list of artifacts to import only additively
+        include_destructive_artifacts: list of artifacts to import destructively
+        fail_if_not_present: flag to fail project if not already present in instance
+        asynchronous: flag to run function asynchronously
+        overwrite_existing: flag to overwrite existing project artifacts
+
+    Returns:
+        operation for project import api call
+    """
+    # initializing empty lists if any
+    if exclude_artifacts is None:
+        exclude_artifacts = []
+    if include_additive_artifacts is None:
+        include_additive_artifacts = []
+    if include_destructive_artifacts is None:
+        include_destructive_artifacts = []
+
+    # check version compatibility for project movement
+    utils.version.enforce_after_or_equal(client=tamr_client, compare_version="2021.005.0")
+
+    # make project import api request
+    body = {
+        "projectArtifact": project_artifact_path,
```

Comment:
Have you tested usage against projects that have spaces in their name? I think the artefact path here might require encoding with `%20` for spaces. 

One way to account for this is wrapping in `urllib.request.pathname2url`.

Reply:
Sounds good, thanks Timo; will test this soon.

Reply:
Just tested and confirmed it needs an encoding. 

```json
{
  "status": 400,
  "class": "javax.ws.rs.BadRequestException",
  "message": "Illegal character in path at index 29: /home/ubuntu/snapshot/Country Code Standardiser-1623667832395.zip"
...
}
```
where index 29 is the space. Only replacing spaces with %20 fixes the issue, but could still result in other issues for different characters.

This would affect all projects that have a space in their name.

Reply:
Discussed this with @pareantoine, @skalish. The decision was to include a workaround for this. Can be part of this PR or a separate parallel PR. 

At the same time will request this be fixed also in product. @tiems90 can you file a SUP ticket for this and link to this one?

The workaround should have a plan for backwards compatibility once the SUP is resolved

Reply:
As suggested by @tiems90, using `urllib.request.pathname2url()` is likely the safest way to do this.  When the necessary fix is incorporated into product, a new PR can be made to only do this when the software version is < the fix version.

Reply:
Tried urllib.request.pathname2url(project_artifact_path) in import body, after exporting a project with spaces in name: '/home/ubuntu/tamr/projectExports/minimal schema mapping with spaces in name-1625100396077.zip'

attempting the import api urllib.request.pathname2url(project_artifact_path) gives: 
'Aborting import. The following validation errors are found:\nProject artifact /home/ubuntu/tamr/projectExports/minimal%2520schema%2520mapping%2520with%2520spaces%2520in%2520name-1625100396077.zip does not exist\n'

It seems like it does want the spaces in the path name. 
I'm not sure if I'm missing something, or if we did different things. 
Thoughts?

Reply:
`%25` is the encoding for `%`, so something strange does seem to be going on here.  Is it possible that typing the `%20` for spaces is necessary in swagger, but passing the string as part of a Python request does not require this?

Reply:
Essentially what I mean is that in order to get `%2520` from a space, something like `urllib.request.pathname2url` must be getting applied twice before the request makes it to the server.  @Tamiro2019 can you check if there's anyway this is happening?  Maybe the path with spaces already replaced is what is coming in as input to your function.

Reply:
found the issue:
the tests were handling sequential export and import tasks, exports were already handling spaces, so imports where double handling them (hence %2520). Thanks for the observation Sam.
I've fixed this by wrapping with urllib.request.pathname2url() only when it detects spaces in the input string (not in general to avoid double wrapping effect, e.g. if the user already encoded the string before calling the import function).